pandas/core/dtypes/concat.py

```diff
@@ -541,36 +541,37 @@ def _concat_rangeindex_same_dtype(indexes):
     """
     from pandas import Int64Index, RangeIndex
 
-    start = step = next = None
+    start = step = next_ = None
 
     # Filter the empty indexes
     non_empty_indexes = [obj for obj in indexes if len(obj)]
 
     for obj in non_empty_indexes:
+        rng = obj._range  # type: range
```

Comment:
I'm a little surprised the annotation here doesn't throw an error - is there an actual type you can put in here?

Reply:
A ``range`` is actually a type in Python 3 :-). So ``issubclass(range, typing.Sequence)`` evaluates to True.